### PR TITLE
[Actions] Created a workflow to move unit tests off of CircleCI

### PR DESCRIPTION
**Pull request by EvanBacon**
CircleCI is not as free as Github Actions. 

I've opted to manually create a test for each package just to keep the migration consistent. The script has the ability to automatically find packages in the future.

### CODE REVIEW

File: .github/workflows/packages.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: yarn install --frozen-lockfile
+      - run: yarn bootstrap
+        env:
+          CI: true
+      - name: Test @expo/dev-tools
+        run: cd packages/dev-tools && ../../scripts/module-scripts-check-packages.sh
```
Comment (ide) [inline .github/workflows/packages.yml]:
```suggestion
        run: cd packages/dev-tools && "$GITHUB_WORKSPACE/scripts/module-scripts-check-packages.sh"
```

File: scripts/module-scripts-check-packages.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+# Copyright 2019-present 650 Industries. All rights reserved.
+
+# exit when any command fails
+set -e
```
Comment (ide) [inline scripts/module-scripts-check-packages.sh]:
```suggestion
set -euo pipefail
```

File: scripts/module-scripts-check-packages.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+# Copyright 2019-present 650 Industries. All rights reserved.
+
+# exit when any command fails
```
Comment (ide) [inline scripts/module-scripts-check-packages.sh]:
```suggestion
```

**Comment by EvanBacon** [comment]
`yarn` and `yarn bootstrap` both seem to take twice as long as Circle CI, we should try and get those to at least be the same duration or faster

**Comment by ide** [comment]
My guess is it's the cache. If you want to email GitHub support about it or tweet at them, go ahead. I think it'd be a pretty meaningful addition to Actions.

**Comment by byCedric** [comment]
> My guess is it's the cache. If you want to email GitHub support about it or tweet at them, go ahead. I think it'd be a pretty meaningful addition to Actions.

I also pinged them related to the Expo GH Action. If they reply with information relevant to you guys too, I'll comment it here. 😄 In my opinion, this is almost a required feature for most large projects.

**Comment by byCedric** [comment]
> > My guess is it's the cache. If you want to email GitHub support about it or tweet at them, go ahead. I think it'd be a pretty meaningful addition to Actions.
> 
> I also pinged them related to the Expo GH Action. If they reply with information relevant to you guys too, I'll comment it here. 😄 In my opinion, this is almost a required feature for most large projects.

TL;DR; They are working on it and hoping to release this feature before the end of the year.

**Comment by brentvatne** [comment]
discussed with @ide - there are a few blockers for merging this so we're going to close it for now.

- github action per-user pricing model is tricky for us 🤑 
- no artifact caching yet, so it's slow
- we can decide what to do when caching is available

**Comment by byCedric** [comment]
Caching is [sort-of here](https://github.com/actions/cache), but you can't download the cache from outside GitHub I think.
